backend/courses/util.py

```diff
@@ -712,3 +712,56 @@ def get_semesters(semesters: str = None) -> list[str]:
             if s not in possible_semesters:
                 raise ValueError(f"Provided semester {s} was not found in the db.")
     return sorted(semesters)
+
+
+def historical_year_probability(current, courses):
+    """
+    current: the current semester represented in the 20XX(A|B|C) format
+    courses: a list of Course objects sorted by date in ascending order
+    Returns a list of 3 probabilities representing the likelihood of taking a course in each
+    semester
+    """
+    prob_distribution = [0.4, 0.3, 0.15, 0.1, 0.05]
+
+    def normalize_and_round(prob, i):
+        """Modifies the probability distribution to account for the
+        fact that the last course was taken i semesters ago."""
+        truncate = prob_distribution[:i]
+        total = sum(truncate)
+        return list(map(lambda x: round(x / total, 3), truncate))
+
+    def get_semester_and_course_index(semester):
+        """Returns an integer representing the semester."""
+        semester_letter = semester[-1]
+        semester_number = 0
+        if semester_letter == "A":
+            semester_number = 1
+        elif semester_letter == "B":
+            semester_number = 2
+        elif semester_letter == "C":
+            semester_number = 3
+        semester_year = int(semester[:-1])
+        return 10 * semester_year + semester_number
+
+    current_index = int(translate_semester(current)) // 10
+    min_index = current_index - 60
+    max_index = current_index - 10
+    p = [0, 0, 0]
+    if courses == []:
+        return p
+    else:
+        last_index = int(translate_semester(courses[0].semester)) // 10
+        if last_index > min_index:
+            prob_distribution = normalize_and_round(
+                prob_distribution, ((current_index - last_index) + 9) // 10
+            )
+    for c in courses:
+        """Provides calculation"""
```

Comment:
Nit: this doc string is not super useful because the code already indicates this